client/tracks/index.js

```diff
@@ -21,6 +21,9 @@ function isEnabled() {
  * @param {Object} [eventProperties] Event properties (optional).
  */
 function recordEvent( eventName, eventProperties ) {
+	// Add `is_test_mode` property to every event.
+	eventProperties.is_test_mode = wcpaySettings?.testMode;
```

Comment:
Yes, it could be not populated, that's why I added the optional chaining operator. And I thought about falling back to false, but I think it's better to just keep it as undefined, to make it clear that we don't know it at that time. WDYT?

Reply:
Ah, of course! I forget that the chaining operator will evaluate to false.

Yes, I think what you said makes sense - setting it to `undefined` if we don't know seems the best option. I guess we'd need to check if passing undefined causes any error with the tracking (I'm assuming it will just accept the value, but I'm also paranoid 😂).

Happy to go ahead with whichever you decide!

Reply:
The JS `undefined` will end being an empty value on Tracks, so when we build a funnel we will be able to choose between "empty", true, or false.

The _good_ thing about `undefined` is that it works as a way to clear/delete definitions. If you run `JSON.stringify` against `{x: 1, y: undefined}` it will evaluate to `{"x": 1}` 🙂

Reply:
Perfect! That sounds like the way to go then. 🙂